`test_lazy.py::test_apply()` test failure

Pulled the latest master (d26083a4752712369804f36e5c8523057d230477) / ran cargo clean, removed venv and recreated from scratch. Running `make test-all`:

```python
___________________________________________________________________________ test_apply ___________________________________________________________________________
[gw1] darwin -- Python 3.8.13 /Users/ivan/dev/rust/polars/py-polars/venv/bin/python

    def test_apply() -> None:
        ldf = pl.LazyFrame({"a": [1, 2, 3], "b": [1.0, 2.0, 3.0]})
        new = ldf.with_columns(pl.col("a").map(lambda s: s * 2).alias("foo"))

        expected = ldf.clone().with_columns((pl.col("a") * 2).alias("foo"))
        assert_frame_equal(new, expected)
        assert_frame_equal(new.collect(), expected.collect())

        for strategy in ["thread_local", "threading"]:
            ldf = pl.LazyFrame({"a": [1, 2, 3] * 20, "b": [1.0, 2.0, 3.0] * 20})
            new = ldf.with_columns(
                pl.col("a").apply(lambda s: s * 2, strategy=strategy).alias("foo")  # type: ignore[arg-type]
            )

            expected = ldf.clone().with_columns((pl.col("a") * 2).alias("foo"))
>           assert_frame_equal(new.collect(), expected.collect())

tests/unit/test_lazy.py:83:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <polars.LazyFrame object at 0x11C82CF10>

    def collect(
        self,
        *,
        type_coercion: bool = True,
        predicate_pushdown: bool = True,
        projection_pushdown: bool = True,
        simplify_expression: bool = True,
        no_optimization: bool = False,
        slice_pushdown: bool = True,
        common_subplan_elimination: bool = True,
        streaming: bool = False,
    ) -> pli.DataFrame:
        """
        Collect into a DataFrame.

        Note: use :func:`fetch` if you want to run your query on the first `n` rows
        only. This can be a huge time saver in debugging queries.

        Parameters
        ----------
        type_coercion
            Do type coercion optimization.
        predicate_pushdown
            Do predicate pushdown optimization.
        projection_pushdown
            Do projection pushdown optimization.
        simplify_expression
            Run simplify expressions optimization.
        no_optimization
            Turn off (certain) optimizations.
        slice_pushdown
            Slice pushdown optimization.
        common_subplan_elimination
            Will try to cache branching subplans that occur on self-joins or unions.
        streaming
            Run parts of the query in a streaming fashion (this is in an alpha state)

        Returns
        -------
        DataFrame

        Examples
        --------
        >>> df = pl.DataFrame(
        ...     {
        ...         "a": ["a", "b", "a", "b", "b", "c"],
        ...         "b": [1, 2, 3, 4, 5, 6],
        ...         "c": [6, 5, 4, 3, 2, 1],
        ...     }
        ... ).lazy()
        >>> df.groupby("a", maintain_order=True).agg(pl.all().sum()).collect()
        shape: (3, 3)
        ┌─────┬─────┬─────┐
        │ a   ┆ b   ┆ c   │
        │ --- ┆ --- ┆ --- │
        │ str ┆ i64 ┆ i64 │
        ╞═════╪═════╪═════╡
        │ a   ┆ 4   ┆ 10  │
        │ b   ┆ 11  ┆ 10  │
        │ c   ┆ 6   ┆ 1   │
        └─────┴─────┴─────┘

        """
        if no_optimization:
            predicate_pushdown = False
            projection_pushdown = False
            slice_pushdown = False
            common_subplan_elimination = False

        if streaming:
            common_subplan_elimination = False

        ldf = self._ldf.optimization_toggle(
            type_coercion,
            predicate_pushdown,
            projection_pushdown,
            simplify_expression,
            slice_pushdown,
            common_subplan_elimination,
            streaming,
        )
>       return pli.wrap_df(ldf.collect())
E       exceptions.ShapeError: Could not add column. The Series length 56 differs from the DataFrame height: 60

polars/internals/lazyframe/frame.py:1430: ShapeError
==================================================================== short test summary info =====================================================================
FAILED tests/unit/test_lazy.py::test_apply - exceptions.ShapeError: Could not add column. The Series length 56 differs from the DataFrame height: 60
=========================================================== 1 failed, 1950 passed, 2 skipped in 31.77s ===========================================================
make: *** [test-all] Error 1
```

### Reproducible example

```python
`venv/bin/pytest -k test_apply`
```

--- a/py-polars/polars/internals/expr/expr.py
+++ b/py-polars/polars/internals/expr/expr.py
@@ -3396,10 +3396,17 @@ class Expr:
 
                 # create partitions with LazyFrames
                 # these are promises on a computation
-                partitions = [
-                    get_lazy_promise(df[i * chunk_size : i * chunk_size + chunk_size])
-                    for i in range(n_threads)
-                ]
+                partitions = []
+                for i in range(n_threads):
+                    # latest partition can be a little longer
+                    # we increase length of slice
+                    if i == (n_threads - 1):
+                        partition_df = df[
+                            i * chunk_size : i * chunk_size + chunk_size * 10
+                        ]
+                    else:
+                        partition_df = df[i * chunk_size : i * chunk_size + chunk_size]
+                    partitions.append(get_lazy_promise(partition_df))
 
                 out = [df.to_series() for df in pli.collect_all(partitions)]
                 return pli.concat(out, rechunk=False)

